test/testpermissions.cpp

```diff
@@ -216,13 +216,10 @@ private slots:
         currentLocalState = fakeFolder.currentLocalState();
         QVERIFY(currentLocalState.find("readonlyDirectory_PERM_M_/cannotBeRemoved_PERM_WVN_.data"));
         QVERIFY(currentLocalState.find("readonlyDirectory_PERM_M_/subdir_PERM_CK_"));
-        // the subdirectory had delete permissions, so the contents were deleted
-        QVERIFY(!currentLocalState.find("readonlyDirectory_PERM_M_/subdir_PERM_CK_/subsubdir_PERM_CKDNV_"));
+        // the subdirectory had delete permissions, but, it was within the recovered directory, so must also get recovered
+        QVERIFY(currentLocalState.find("readonlyDirectory_PERM_M_/subdir_PERM_CK_/subsubdir_PERM_CKDNV_"));
```

Comment:
we should mention that in the documentation and/or release notes to make sure people expect that change in behavior

Reply:
> we should mention that in the documentation and/or release notes to make sure people expect that change in behavior

Why? There will be explanatory error notifications in the activity list and then, people would end up having 2 folders - renamed (which gets uploaded as a full copy), and an old folder untouched on the server.

But, yes, we can add this into release notes if you want.

Reply:
I will take care of it when sending the changelog to marketing
I want to say something like
to prevent data loss we are now making sure we upload all folders with new files in them even if they had been deleted from server